### DISCUSSION TITLE
Clippy ignore possible truncation with explicit `X.trunc() as Y`

### DISCUSSION THREAD

**Discussion by ShayBox** [rust-lang/rust-clippy#13607] [Ideas]
Clippy should use explicit .trunc() before as to ignore possible truncation.

**Comment by flip1995** [comment]
Are you referring to a specific lint? Can you give a code example, where you want to see different behavior from Clippy?

**Reply by ShayBox** [reply]
`clippy::cast_possible_truncation` could recognize when `.trunc()`, `.floor()`, or `.ceil()` is used before `as X` to disable the warning (at least for floats to integers where the truncation is only floating point precision)

**Reply by flip1995** [reply]
Can you create an issue for this enhancement, please? Discussions are more for asking questions than for feature requests.
